apps/scan/src/main/scala/org/lfdecentralizedtrust/splice/scan/store/AcsSnapshotStore.scala

```diff
@@ -124,28 +124,44 @@ class AcsSnapshotStore(
-                select row_id,
-                       concat(package_name, ':', template_id_module_name, ':', template_id_entity_name),
-                       stakeholder
-                from creates_to_insert
-                         cross join unnest(array_cat(signatories, observers)) as stakeholders(stakeholder)
-                where history_id = $historyId
-                  and migration_id = $migrationId
-                -- consistent ordering across SVs
-                order by created_at, contract_id
-                returning row_id
-        )
+                                                            on contracts.contract_id = creates.contract_id),
+                inserted_rows as (insert into acs_snapshot_data (create_id, template_id, stakeholder)
+                                  select row_id,
+                                         concat(package_name, ':', template_id_module_name, ':', template_id_entity_name),
+                                         stakeholder
+                                  from creates_to_insert
+                                           cross join unnest(array_cat(signatories, observers)) as stakeholders(stakeholder)
+                                  where history_id = $historyId
+                                    and migration_id = $migrationId
+                                  -- consistent ordering across SVs
+                                  order by created_at, contract_id
+                                  returning row_id, create_id, template_id, stakeholder
+                )
         insert
-        into acs_snapshot (snapshot_record_time, migration_id, history_id, first_row_id, last_row_id)
-        select $until, $migrationId, $historyId, min(row_id), max(row_id)
+        into acs_snapshot (snapshot_record_time, migration_id, history_id, first_row_id, last_row_id, unlocked_amulet_balance, locked_amulet_balance)
+        select
+          $until,
+          $migrationId,
+          $historyId,
+          min(inserted_rows.row_id),
+          max(inserted_rows.row_id),
+          sum(case when inserted_rows.template_id = ${PackageQualifiedName
+            .getFromResources(
+              Amulet.TEMPLATE_ID_WITH_PACKAGE_ID
+            )
+            .toString} and stakeholder=$dsoParty then (create_arguments->'record'->'fields'->2->'value'->'record'->'fields'->0->'value'->>'numeric')::numeric else 0 end),
+          sum(case when inserted_rows.template_id = ${PackageQualifiedName
+            .getFromResources(
+              LockedAmulet.TEMPLATE_ID_WITH_PACKAGE_ID
+            )
+            .toString} and stakeholder=$dsoParty then (create_arguments->'record'->'fields'->0->'value'->'record'->'fields'->2->'value'->'record'->'fields'->0->'value'->>'numeric')::numeric else 0 end)
         from inserted_rows
-        having min(row_id) is not null;
+        join creates_to_insert on inserted_rows.create_id = creates_to_insert.row_id
```

Comment:
I guess filtering by templates here isn't actually faster than just having the filter in sum since we d't have an index on the materialized view?